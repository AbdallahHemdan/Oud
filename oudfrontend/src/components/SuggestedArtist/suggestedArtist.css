.CenterText {
  color: white;
  font-size: 200%;
  text-align: center;
}

.FText {
  margin: 0 auto;
  left: 0;
  right: 0;
  display: block;
  align-items: center;
  align-self: center;
  text-align: center;
}

.navtree {
  margin: 0 auto;
  left: 0;
  right: 0;
  display: block;
}

.SuggNav {
  position: fixed;
  width: 100%;
  top: 0;
  margin-bottom: 200%;
  background-color: black;
  z-index: 9999;
}

.spacc {
  margin-bottom: 300px;
}

.SuggNav .FText {
  margin: 0 auto;
  left: 0;
  right: 0;
  margin-bottom: 20px;
}

.SuggLogo {
  width: 150px;
  margin: 0 auto;
  left: 0px;
  right: 0px;
  transition: 0.5s ease;
}

.NavBarStyle_bg-custom {
  background-color: rgba(0, 0, 0, 0.6);
  box-shadow: 0 3px 8px -2px rgba(240, 235, 222, 0.1);
}

.navContants .CenterText {
  text-align: center;
}

.subText {
  color: white;
}

.NavBarStyle_navbar-expand-lg {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 9999;
  width: 100%;
  background-color: black;
}

.NavBarStyle_navbar-brand {
  text-align: left;
}

.SearchInp {
  border-radius: 50px;
  margin: 0 auto;
  left: 0;
  right: 0;
  position: relative;
  width: 110% !important;
  height: 50px;
  box-shadow: none;
}

.SearchInp:focus {
  border: 15px;
  color: black;
  font-weight: bold;
}

.SuggFooter {
  background-color: black;
  color: aliceblue;
  font-weight: bold;
  position: fixed;
  margin: 0 auto;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 90px;
  text-align: center;
}

.FinishBtn {
  width: 20%;
  height: 50px;
  margin: 0 auto;
  left: 0;
  right: 0;
  display: block;
  color: white;
  font-size: 15px;
  font-weight: bold;
  background-color: goldenrod;
  border-radius: 50px;
  border-color: goldenrod;
}

.ArtistImage {
  width: 150px;
  height: 200px;
  border-radius: 100%;
  opacity: 1;
  display: block;
  transition: 0.5s ease;
  backface-visibility: hidden;
  text-align: center;
}

.ArtistText {
  justify-content: center;
  justify-items: center;
  text-align: center;
  margin-top: -50px;
  color: white;
  font-size: larger;
  font-weight: bold;

}

.ArtistName {
  font-size: 20px;
  font-weight: bold;
}

.ArtistImage .ArtistName {
  display: block;
}

.hide {
  opacity: 0;
  transition: opacity 0.5s ease-out;
}

.show {
  opacity: 1;
  transition: opacity 0.5s ease-out;
}

.hideLogo {
  visibility: hidden;
  opacity: 0;
  transition: visibility 0s, opacity 0.5s linear;
}

.dis {
  display: none;
  transition: opacity 1s ease-out;
  opacity: 0;
}

.half {
  opacity: 100%;
  transition: 1.5s ease;
  background-color: goldenrod;
  border-radius: 100%;
}

.textTop {
  top: 0;
}

.ArtistCard {
  margin: 0;
  padding: 0;
}

.ArtistCard:hover {
  background: goldenrod;
  border-radius: 50%;
}

.ArtistCard:not(:hover) {
  background: transparent;
}

.ArtistMap {
  width: auto;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
  box-sizing: border-box;
}

.MainArtistC {
  margin: 0 auto;
  left: 0;
  right: 0;
  display: block;
  margin-left: 50px;
  margin-right: 50px;
}

.Artists {
  align-items: center;
  align-self: center;
  margin: 0 auto;
  left: 0;
  right: 0;
  display: block;
  margin-top: 30px;
}

.heart {
  width: 10%;
  border-radius: 500%;
  height: auto;
  display: block;
  backface-visibility: hidden;
  position: absolute;
  width: 40.8%;
  opacity: 0;
  transition: 0.5s ease;
  background-color: goldenrod;
  border-radius: 100%;
  z-index: 999;
}

@media (max-width: 1200px) {
  .ArtistMap {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
  }

  .clickedPic {
    transform: translate(0, -11rem) !important;
  }
}

@media (max-width: 1100px) {
  .clickedPic {
    transform: translate(0, -10rem) !important;
  }
}

@media (max-width: 1000px) {
  .clickedPic {
    transform: translate(0, -9rem) !important;
  }
}

@media (max-width: 900px) {
  .clickedPic {
    transform: translate(0, -8rem) !important;
  }
}

@media (max-width: 800px) {
  .ArtistMap {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }

  .clickedPic {
    transform: translate(0, -9rem) !important;
  }
}

.spaceeee {
  padding: 4%;
}

.SingleCard .clickedPic {
  align-content: center;
  margin: 0 auto;
}

.showbas {
  background-color: rgb(35, 33, 28);
  border-radius: 100%;
}

.asd {
  opacity: 1;
  background-color: goldenrod;
  border-radius: 100%;
  transition: opacity 1s ease-out;
}

.TextFotter {
  text-align: center;
  margin-top: 20px;
}

.s {
  color: white;
  width: 10%;
  border-radius: 100%;
  font-size: 20px;
  position: absolute;
  -webkit-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  text-align: center;
  background-color: rgb(247, 215, 136);
}

.clickedPic {
  width: 4rem;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  transform: translate(0, -200%);
}

.circular--portrait {
  position: relative;
  width: 100%;
  height: 250px;
  overflow: hidden;
  border-radius: 50%;
}

.circular--portrait img {
  width: 100%;
  height: auto;
}