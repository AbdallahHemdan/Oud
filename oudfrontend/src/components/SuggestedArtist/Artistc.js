import React, {Component} from 'react';
import ArtistCards from './Artistscards';
import './suggestedArtist.css';
import {base, subUrl, prodUrl} from './../../config/environment';

/**
 * @function
 * @param {object} props
 * @returns {JSX}
 */
export default class Artistc extends Component {
  constructor(props) {
    super(props);
    this.state = {artists: this.props.artists};
  }

  render = () => {
    const subPath = base === prodUrl ? subUrl : '';
    return (
      <div className="Artists">
        <div className="row ArtistMap">
          {this.props.artists.map((
            element,
            index /** here i map the artist and
        send the props the name and the display name */
          ) => (
            <ArtistCards
              displayName={element.displayName}
              image={`${subPath}${element.images[0]}`}
              key={index}
              handleSelect={this.props.handleSelect}
              handleRelated={this.props.handleRelated}
              id={element.id}
            />
          ))}
        </div>
      </div>
    );
  };
}
