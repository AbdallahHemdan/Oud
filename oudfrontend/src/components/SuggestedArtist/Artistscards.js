import React, {Component} from 'react';
import './suggestedArtist.css';
import loved from './../../assets/images/icons/H.png';
import Axios from 'axios';

/**
 *@class 
 the artist card that have the structure of the one card 
 */
class ArtistCard extends Component {
  /**
   * constructor
   * @param Toggle css
   */
  constructor(props) {
    super(props);
    this.state = {
      Toggle: true,
      selected: false,
      isSelected: false,
      displayName: this.props.displayName,
      image: this.props.image,
      id: this.props.id,
    };
  }
  handleSelect = () => {
    this.setState({
      selected: !this.state.selected,
      isSelected: !this.state.isSelected,
    });
    this.props.handleSelect(this.props.id, this.state.selected);
    this.props.handleRelated(this.props.id);
  };

  toggleSelected = () => {
    if (this.state.selected === false) this.setState({selected: true});
    else this.setState({selected: false});
  };
  render() {
    return (
      <div
        className="container"
        onMouseEnter={() => this.setState({Toggle: false})}
        onMouseLeave={() => this.setState({Toggle: true})}
        onClick={this.handleSelect}
      >
        <div className="SingleCard">
          <div className="">
            <div className={`${this.state.Toggle ? 'show' : 'half'}`}>
              <div className={this.state.isSelected ? 'asd' : ''}>
                <div className="circular--portrait">
                  <img
                    alt=""
                    src={this.state.image}
                    style={{
                      width: '100%',
                      height: '100%',
                    }}
                    data-testid="artistImage"
                    className={`ArtistImage ${
                      this.state.Toggle || this.state.isSelected
                        ? 'show'
                        : 'half '
                    }`}
                  ></img>
                </div>
              </div>
            </div>
            <img
              alt=""
              src={loved}
              className={`clickedPic  ${this.state.selected ? 'show' : 'hide'}`}
            ></img>
          </div>
          <p className="ArtistText">{this.state.displayName}</p>
        </div>
      </div>
    );
  }
}
export default ArtistCard;
