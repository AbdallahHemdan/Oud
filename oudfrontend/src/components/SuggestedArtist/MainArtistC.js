import React, {Component} from 'react';
import adele from './../../assets/images/adeleImg.png';
import ArtistC from './Artistc';
import './suggestedArtist.css';
import axios from 'axios';
import {base, subUrl, prodUrl} from './../../config/environment';
import {config} from './../../utils/auth';

const fetchUserInfo = `${base}/artists/some`;
class MainArtistC extends Component {
  constructor(props) {
    super(props);
    this.state = {
      artists: [],
    };
    axios
      .get(fetchUserInfo, config)
      .then((result) => {
        console.log('result', result.data);
        this.setState({artists: result.data});
      })
      .catch((err) => {
        console.log('error', err.result);
      });
  }
  check = (rA, relatedArtist) => {
    console.log(rA.id !== relatedArtist.id);
    return rA.id !== relatedArtist.id;
  };

  handleRelated = (id) => {
    axios
      .get(`${base}/artists/${id}/related-artists`)
      .then((response) => {
        let relatedArtists = response.data.artists;
        let toBeDeleted = [];
        relatedArtists.forEach((relatedArtist) => {
          this.state.artists.forEach((artist) => {
            if (artist.id === relatedArtist.id) {
              toBeDeleted.push(relatedArtist);
            }
          });
        });
        relatedArtists = relatedArtists.filter(
          (a) => toBeDeleted.indexOf(a) === -1
        );
        // let cutIndex = 0;
        // for (let i = 0; i < this.state.artists.length; i++) {
        //   if (this.state.artists[i].id === id) {
        //     cutIndex = i;
        //     break;
        //   }
        // }
        // let artists = this.state.artists;
        // let after = artists.splice(cutIndex + 1);
        // console.log(cutIndex, artists, after);
        this.setState({
          artists: this.state.artists.concat(relatedArtists),
        });
      })
      .catch((err) => {
        console.log(err.response);
      });
  };
  render() {
    return (
      <div className="MainArtistC">
        <div className="spacc"></div>
        <ArtistC
          artists={this.state.artists}
          handleSelect={this.props.handleSelect}
          handleRelated={this.handleRelated}
        ></ArtistC>
        <div className="spaceeee"></div>
      </div>
    );
  }
}
export default MainArtistC;
