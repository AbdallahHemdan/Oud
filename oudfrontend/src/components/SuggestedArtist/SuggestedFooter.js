import React, {Component} from 'react';
// import {Link} from 'react-router-dom';
import './suggestedArtist.css';
import Axios from 'axios';
import {base, subUrl, prodUrl} from '../../config/environment';
import {config} from '../../utils/auth';

const artist = `${base}/me/following?type=artist`;

class SuggestedFooter extends Component {
  constructor(props) {
    super(props);
    this.state = {selectedIDS: this.props.selectedIds};
  }

  handelOnClick = () => {
    let ids = this.state.selectedIDS;
    Axios.put(artist, {ids}, config)
      .then((response) => {
        console.log('res', response);
        window.location = '/';
      })
      .catch((error) => {
        console.log(error.response);
      });
  };

  render() {
    return this.props.selected() ? (
      <div className="SuggFooter">
        <footer className="page-footer font-small unique-color-dark pt-4">
          <div className="container">
            <button
              data-testid="FooterBtn"
              className="FinishBtn"
              onClick={this.handelOnClick}
            >
              Finish
            </button>
          </div>
        </footer>
      </div>
    ) : (
      <div className="SuggFooter">
        <footer className="page-footer ">
          <div className="container TextFotter" data-testid="Footertext">
            <p>Select at least one artist or use the search bar.</p>
          </div>
        </footer>
      </div>
    );
  }
}

export default SuggestedFooter;
