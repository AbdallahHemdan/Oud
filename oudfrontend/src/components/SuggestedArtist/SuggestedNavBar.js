import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import logo from '../../assets/images/Logoc.png';
import './suggestedArtist.css';
/**
 * @class
 */
class SuggestedNavBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      scrolled: false,
    };
  }

  componentDidMount = () => {
    window.addEventListener('scroll', () => {
      const isTop = window.scrollY < 80;
      if (isTop !== true) {
        this.setState({scrolled: true});
      } else this.setState({scrolled: false});
    });
  };

  render() {
    return (
      <div className={`SuggNav ${this.state.scrolled ? 'navbarco' : ''}`}>
        <nav className=" navbar navbar-expand-lg  bg-custom NavBarStyle_bg-custom navtree">
          <div className="  navbar-nav ml-auto  font-weight-bold ">
            <div className="FText">
              <Link to="/welcome" className="navbar-brand ">
                <img
                  className={`img-responsive SuggLogo ${
                    this.state.scrolled ? 'hideLogo dis' : 'show'
                  }`}
                  src={logo}
                  alt="Oud logo"
                  data-testid="oudlogoNvaBarSugg"
                  id="oudlogo"
                />
              </Link>
              <p
                data-testid="TextNvaBarSugg"
                className={`CenterText  ${
                  this.state.scrolled ? 'textTop' : 'show'
                }`}
              >
                Let's get to know each other!
              </p>
              <p data-testid="TextNvaBarSugg" className="subText container">
                What are your favorite artists? The more you select, the better
                your recommendations!
              </p>
              {/* <form className="form-inline container">
                <input
                  className="form-control SearchInp"
                  type="search"
                  placeholder="Search"
                  aria-label="Search"
                />
              </form> */}
            </div>
          </div>
        </nav>
      </div>
    );
  }
}

export default SuggestedNavBar;
