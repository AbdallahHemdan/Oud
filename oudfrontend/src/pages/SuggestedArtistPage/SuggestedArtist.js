import React, {Component} from 'react';
import SuggestedNavBar from './../../components/SuggestedArtist/SuggestedNavBar';
import SuggestedFooter from './../../components/SuggestedArtist/SuggestedFooter';
import MainArtistC from './../../components/SuggestedArtist/MainArtistC';
export default class SuggestedArtist extends Component {
  constructor(props) {
    super(props);
    this.state = {isSelected: false, selectedIds: []};
  }
  check = (sId, id) => {
    return sId !== id;
  };
  componentDidUpdate() {}
  handleSelect = (id, remove) => {
    let ids = this.state.selectedIds;
    if (!remove) {
      ids.push(id);
      this.setState({selectedIds: ids});
    } else {
      const selectedIds = this.state.selectedIds.filter((sId) =>
        this.check(sId, id)
      );
      this.setState({selectedIds});
    }
  };
  isSelected = () => {
    return this.state.selectedIds.length;
  };
  onClick = (event) => {
    this.setState({isSelected: event.target.value});
    if (event.target.value) {
      event.target.value = !event.target.value;
    }
  };
  handelRenderComp = () => {
    if (this.SuggestedFooterB.state.choo === true) {
    }
  };
  render() {
    return (
      <div>
        <SuggestedNavBar />
        <MainArtistC handleSelect={this.handleSelect} />
        <SuggestedFooter
          selected={this.isSelected}
          selectedIds={this.state.selectedIds}
        />
        )}
      </div>
    );
  }
}
