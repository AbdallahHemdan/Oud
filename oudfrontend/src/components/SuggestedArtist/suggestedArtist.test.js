import React from 'react';
import {shallow, mount} from 'enzyme';
import * as renderer from 'react-test-renderer';
import SuggestedNavBar from './SuggestedNavBar';
import SuggestedArtist from './../../pages/SuggestedArtistPage/SuggestedArtist';
import ArtistCard from './Artistscards';

const setup = () => {
  return shallow(<SuggestedNavBar />);
};
describe('Test the the Suggested artist page ', () => {
  describe('test the render of the navbar ', () => {
    it('test the Logo render ', () => {
      const component = shallow(<SuggestedNavBar />);
      const logo = component.find(`[data-testid="oudlogoNvaBarSugg"]`);
      expect(logo.length).toBe(1);
    });
    it('render the text of the navbar ', () => {
      const component = shallow(<SuggestedNavBar />);
      const text = component.find(`[data-testid="TextNvaBarSugg"]`);
      expect(text.length).toBe(2);
    });
  });
  describe('text the render of the footer of the navbar on the suggested artist', () => {
    let component;
    beforeEach(() => {
      component = setup();
    });
    // it('test the render of the conditional rendering of the footer ', () => {
    //   const buttoncomp = shallow(<SuggestedFooterB />);
    //   const textcomp = shallow(<SuggestedFooterT />);
    //   expect(buttoncomp.state().choo).toBeTruthy();
    //   expect(textcomp.state().choo).toBeFalsy();
    // });
  });
  describe('Test the Artist card ', () => {
    it('test the image click', () => {
      const wrapper = shallow(<ArtistCard />);
      const image = wrapper.find(`[data-testid="artistImage"]`);
      image.simulate('click');
      expect(wrapper.state().isSelected).toBeFalsy();
    });
  });

  describe('test the  a snapshot', () => {
    it('test the snapshot ', () => {
      const wrapper = shallow(<SuggestedArtist />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
