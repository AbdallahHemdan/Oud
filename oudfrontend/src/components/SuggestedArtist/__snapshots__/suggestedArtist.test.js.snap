// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the the Suggested artist page  test the  a snapshot test the snapshot  1`] = `ShallowWrapper {}`;
